core/services/keeper/upkeep_executer.go

```diff
@@ -24,40 +24,50 @@ const (
 )
 
 func NewUpkeepExecutor(
+	job job.Job,
 	db *gorm.DB,
 	ethClient eth.Client,
+	headRelayer *services.HeadRelayer,
 ) *UpkeepExecutor {
 	return &UpkeepExecutor{
-		blockHeight:    atomic.NewInt64(0),
-		wgDone:         sync.WaitGroup{},
+		chStop:         make(chan struct{}),
```

Comment:
Not worth blocking the PR on, but would be a little more tidy if this line was right above the `wgDone` line